test/homepage-noweb-test.js

```diff
@@ -41,24 +40,31 @@ suite.addBatch({
         },
         "and we visit the root URL": {
             topic: function() {
-                var browser;
-                browser = new Browser();
+                var cb=this.callback,
+                    browser = new Browser();
 
-                browser.visit("http://localhost:4815/", this.callback);
+                browser.visit("http://localhost:4815/", function(){
+                        cb(!browser.success, browser);
+                });
+            },
+            teardown: function(br){
+                if(br && br.window.close){
+                    br.window.close();
+                }
             },
             "it works": function(err, br) {
                 assert.ifError(err);
-                assert.isTrue(br.success);
+		        br.assert.success();
             },
             "it has the right title": function(err, br) {
                 assert.ifError(err);
-                assert.isTrue(br.success);
-                assert.equal(br.text("title"), "API - pump.io");
+		        br.assert.success();
+                br.assert.text("title", "API - pump.io");
```

Comment:
I haven't looked closely at the actual code in this section since the whitespace makes the diff unreadable. Easier to just wait :)

(Mostly writing this comment so _I_ don't forget).